### PR TITLE
Rename GStrPtr to GStringPtr

### PR DESCRIPTION
**Pull request by pbor**
Since this is an owned type this is more consistent.

### CODE REVIEW

**Comment by jf2048** [comment]
If you want there could also be a `to_str` that just calls `to_gstr().as_str()`

**Comment by pbor** [comment]
> If you want there could also be a `to_str` that just calls `to_gstr().as_str()`

yeah, I opted for the minimal public API, but I can add it if that is preferred

**Comment by jf2048** [comment]
It's up to you, either way is fine :)

File: glib/src/collections/strv.rs [hunk 1]
```diff
@@ -201,12 +201,12 @@ impl IntoIter {
     // rustdoc-stripper-ignore-next
     /// Returns the remaining items as slice.
     #[inline]
-    pub fn as_slice(&self) -> &[GStrPtr] {
+    pub fn as_slice(&self) -> &[GStringPtr] {
```
Comment (jf2048) [inline glib/src/collections/strv.rs:204]:
```suggestion
    pub const fn as_slice(&self) -> &[GStringPtr] {
```

File: glib/src/collections/strv.rs [hunk 2]
```diff
@@ -675,14 +678,14 @@ impl StrV {
     }
 
     // rustdoc-stripper-ignore-next
-    /// Borrows this slice as a `&[GStrPtr]`.
+    /// Borrows this slice as a `&[GStringPtr]`.
     #[inline]
-    pub fn as_slice(&self) -> &[GStrPtr] {
+    pub fn as_slice(&self) -> &[GStringPtr] {
```
Comment (jf2048) [inline glib/src/collections/strv.rs:683]:
```suggestion
    pub const fn as_slice(&self) -> &[GStringPtr] {
```

**Review by jf2048 — CHANGES_REQUESTED** [review]

**Comment by sdroege** [comment]
> yeah, I opted for the minimal public API, but I can add it if that is preferred

Well, doesn't cost much so why not add it? :)

File: glib/src/collections/strv.rs [hunk 3]
```diff
@@ -443,11 +446,11 @@ impl StrV {
         if len == 0 {
             StrV::default()
         } else {
-            // Need to fully copy the array here.
-            let slice = Self::from_glib_borrow_num(ptr, len);
-            let mut s = Self::new();
-            s.extend_from_slice(slice);
-            s
+            StrV {
+                ptr: ptr::NonNull::new_unchecked(ffi::g_strdupv(ptr as *mut _)),
```
Comment (sdroege) [inline glib/src/collections/strv.rs:450]: The reason for the previous implementation was that we know the length while `g_strdupv()` first needs to determine the length before copying

**Review by sdroege — COMMENTED** [review]
Looks good otherwise, thanks :)

File: glib/src/collections/strv.rs (hunk 3 above)
Comment (pbor) [reply glib/src/collections/strv.rs:450]: Yes, but `extend_from_slice` requires `AsRef` so it cannot be used anymore, I guess I can re-implement it

Comment (sdroege) [reply glib/src/collections/strv.rs:450]: Ah I see. Well, worst case start with `g_memdup()` :)

Comment (pbor) [reply glib/src/collections/strv.rs]: I need to copy all the strings as well, so memdup does not sound the right thing... I simply allocated and then copied all the strings

**Review by sdroege — APPROVED** [review]

**Comment by sdroege** [comment]
Also fixed gtk4-rs for the rename.

**Pull request merged**